File: go/internal/booking-service/core/domain/entities/guest.entity.go
package entities

import (
	valueobjects "book/internal/booking-service/core/domain/value-objects"
)

type Guest struct {
	Id       uint                `gorm:"primaryKey"`
	Name     string              `json:"name"`
	Surname  string              `json:"surname"`
	Email    string              `json:"email"`
	Document valueobjects.Person `gorm:"embedded"`
}

File: nestjs/src/booking/core/domain/entities/guest.entity.ts
import { Entity } from "@/shared/domain/entities/entity";
import { Person } from "../value-objects/person.valueobject";

export type GuestProps = {
    id: number;
    name: string;
    surname: string;
    email: string;
    document: Person;
}

export default class GuestEntity extends Entity<GuestProps> {
    constructor(public readonly props: GuestProps){
        super(props);
    }
}